lib/management/patchConfiguration.js

```diff
@@ -0,0 +1,89 @@
+const arsenal = require('arsenal');
+const config = require('../../conf/Config');
+
+const USERSBUCKET = arsenal.constants.usersBucket;
+const PENSIEVE = 'PENSIEVE';
+const METASTORE = '__metastore';
+
+function getIngestionBuckets(metadata, logger, cb) {
+    const mongoClient = metadata.getClient();
+    const m = mongoClient.getCollection(METASTORE);
+    m.find({
+        '_id': {
+            $nin: [PENSIEVE, USERSBUCKET],
+        },
+        'value.ingestion': {
+            $type: 'object',
+        },
+    }).project({
+        'value.name': 1,
+        'value.ingestion': 1,
+        'value.locationConstraint': 1,
+    }).toArray((err, doc) => {
+        if (err) {
+            logger.error('error getting ingestion buckets',
+                { error: err.message });
+            return cb(err);
+        }
+        return cb(null, doc.map(i => i.value));
+    });
+}
+
+function updateIngestionBuckets(locations, metadata, logger, cb) {
+    getIngestionBuckets(metadata, logger, (err, buckets) => {
+        if (err) {
+            return cb(err);
+        }
+        config.setIngestionBuckets(locations, buckets);
+        return cb();
+    });
+}
+
+function updateLocations(conf, logger, cb) {
+    try {
+        const locations = conf.locations;
+        const locationsWithReplicationBackend = Object.keys(locations)
+        // NOTE: In Orbit, we don't need to have Scality location in our
+        // destination bootstrapList config, since we do not replicate to
+        // any Scality Instance yet.
+        .filter(key => locations[key].locationType !== 'location-file-v1')
+        .reduce((obj, key) => {
+            /* eslint no-param-reassign:0 */
+            obj[key] = locations[key];
+            return obj;
+        }, {});
+        config.setBootstrapList(locationsWithReplicationBackend);
+
+        Object.keys(locations).forEach(locName => {
+            config.setIsTransientLocation(
+                locName, locations[locName].isTransient);
+        });
+    } catch (error) {
+        logger.error('error updating locations', { error: error.message });
+        return cb(error);
+    }
+    return cb(null, conf.version);
+}
+
+function patchConfiguration(overlayVersion, conf, metadata, logger, cb) {
+    if (conf.version === undefined) {
+        return process.nextTick(cb, null);
+    }
+    if (conf && conf.locations) {
+        // Update ingestion buckets regardless of config overlay version
+        return updateIngestionBuckets(conf.locations, metadata, logger, err => {
```

Comment:
Pass an extra param to `initManagement`. Based on this param, we perform this query.
We should not impact the performance of other services because of this